core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProjectOrchestrator.scala

```diff
@@ -0,0 +1,77 @@
+package org.broadinstitute.dsde.rawls.billing
+
+import akka.http.scaladsl.model.StatusCodes
+import org.broadinstitute.dsde.rawls.dataaccess.{GoogleServicesDAO, SamDAO}
+import org.broadinstitute.dsde.rawls.model.{CreateRawlsV2BillingProjectFullRequest, CreationStatuses, ErrorReport, ErrorReportSource, RawlsBillingProject, SamBillingProjectPolicyNames, SamBillingProjectRoles, SamPolicy, SamResourcePolicyName, SamResourceTypeNames, SamServicePerimeterActions, ServicePerimeterName, UserInfo}
+import org.broadinstitute.dsde.rawls.user.UserService.syncBillingProjectOwnerPolicyToGoogleAndGetEmail
+import org.broadinstitute.dsde.rawls.{RawlsExceptionWithErrorReport, StringValidationUtils}
+import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets.UTF_8
+import scala.concurrent.{ExecutionContext, Future}
+
+
+/**
+ * Knows how to provision billing projects with external cloud providers
+ */
+class BillingProjectOrchestrator(userInfo: UserInfo, samDAO: SamDAO, gcsDAO: GoogleServicesDAO, billingRepository: BillingRepository)(implicit val executionContext: ExecutionContext) extends StringValidationUtils {
```

Comment:
This class is almost a pure extraction of the v2 billing project code from UserService. This https://github.com/broadinstitute/rawls/pull/1961 PR will introduce the Azure+BPM code path and related API client calls, etc.